### PR TITLE
Security Compliance

### PR DESCRIPTION
**Pull request by vvikramb**
Security Compliance

### CODE REVIEW

File: juniper_official/Security-Compliance/check-snmp-server-config.rule [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            }
+            field targetParameterName {
+                sensor snmpserver {
+                    path target-parameters-name;
+                }
+                type string;
+            }
+            /*
+             * Anomaly detection logic.
+             */
```
Comment (gvarao) [inline juniper_official/Security-Compliance/check-snmp-server-config.rule:80]: Rule deployed but no data is coming

File: juniper_official/Security-Compliance/check-ssh-services.rule [hunk 2]
```diff
@@ -0,0 +1,267 @@
+/*
+*
+* Checks ssh services configuration. 
+* 
+*/
+healthbot {
```
Comment (gvarao) [inline juniper_official/Security-Compliance/check-ssh-services.rule:6]: Rule deployed but no data is coming

**Review by gvarao — CHANGES_REQUESTED** [review]

File: juniper_official/Security-Compliance/check-community-access.rule [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                            }
+                            products QFX {
+                                platforms All {
+                                    releases 18.1R1 {
+                                        release-support min-supported-release;
+                                    }
+                                }
+                            }
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-community-access.rule:122]: Added SRX in rule properties.

File: juniper_official/Security-Compliance/check-interface-protocols.rule [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            sensor protocolsInterface {
+                /*
+                 * Sensor configuration to get data from network devices.
+                 */
+                iAgent {
+                    file protocolsInterface.yml;
+                    table ProtocolsTable;
+                    frequency 60s;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-protocols.rule:23]: Changed frequency from 60s to 6h.

File: juniper_official/Security-Compliance/check-interface-protocols.rule [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                            }
+                            products QFX {
+                                platforms All {
+                                    releases 18.1R1 {
+                                        release-support min-supported-release;
+                                    }
+                                }
+                            }
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-protocols.rule:166]: Added SRX in rule properties.

File: juniper_official/Security-Compliance/ntpAuthentication.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+---
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/ntpAuthentication.yml:1]: Updated spaces.

File: juniper_official/Security-Compliance/ntpServers.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/ntpServers.yml:1]: Updated spaces.

File: juniper_official/Security-Compliance/check-community-access.rule [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            keys communityname;
+            description "Checks for snmp community access";
+            sensor communityAccess {
+                /*
+                 * Sensor configuration to get data from network devices.
+                 */
+                iAgent {
+                    file snmpcommunityAccess.yml;
+                    table snmpcommunityAccessTable;
+                    frequency 60s;
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-community-access.rule]: Update frequency from 60s to 6h.

File: juniper_official/Security-Compliance/check-community-access.rule [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                }
+            }
+            /*
+             * Fields defined with variables and constant values
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-community-access.rule]: Updated comments for fields.

File: juniper_official/Security-Compliance/check-community-access.rule [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+            trigger snmpcommunityAccess {
+                frequency 2offset;
+                term communityaccessread {
+                    when {
+                        matches-with "$authorization" read-only;
+                    }
+                    then {
+                        status {
+                            color green;
+                            message "Community $communityname is applied to client list $communityListName and has read only permission.";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-community-access.rule]: Removed community name in message.

File: juniper_official/Security-Compliance/check-community-access.rule [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+                    }
+                }
+                term communityaccess-readwrite {
+                    when {
+                        matches-with "$authorization" read-write;
+                    }
+                    then {
+                        status {
+                            color yellow;
+                            message "Community $communityname is applied to client list $communityListName and  has read-write permission.";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-community-access.rule]: Removed community name in message.

File: juniper_official/Security-Compliance/check-interface-protocols.rule [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+             */
+            description "Checks protocol is not enabled on all interfaces.";
+            sensor protocolsInterface {
+                /*
+                 * Sensor configuration to get data from network devices.
+                 */
+                iAgent {
+                    file protocolsInterface.yml;
+                    table ProtocolsTable;
+                    frequency 60s;
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-protocols.rule]: Changed frequency from 60s to 6h.

File: juniper_official/Security-Compliance/check-interface-protocols.rule [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+            trigger isis_interface_all {
+                frequency 2offset;
+                term isis-interface-all-present {
+                    when {
+                        matches-with "$isisInterfaceName" all;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "isis interface all should not be set.";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-protocols.rule]: Checking if "interface all" is enabled for isis protocols and raising alarm if it is set.

File: juniper_official/Security-Compliance/check-interface-protocols.rule [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+            trigger lldp_interface_all {
+                frequency 2offset;
+                term lldp-interface-all-present {
+                    when {
+                        matches-with "$lldpInterfaceName" all;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "lldp interface all should not be set.";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-protocols.rule]: Checking if "interface all" is enabled for lldp protocols and raising alarm if it is set.

File: juniper_official/Security-Compliance/check-interface-protocols.rule [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+            trigger source_packet_routing {
+                frequency 2offset;
+                term source-packet-routing-present {
+                    when {
+                        matches-with "$source-packet-routing" true;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "source-packet-routing should not be enabled.";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-protocols.rule]: Checking if "source packet routing is enabled" and raising red alarm if it is set.

File: juniper_official/Security-Compliance/check-community-access.rule [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+            }
+            /*
+             * Fields defined with variables and constant values
+             */
+            field authorization {
+                sensor communityAccess {
+                    path authorization;
+                }
+                type string;
+                description "authorization type.";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-community-access.rule]: Removed full stop from all descriptions.

File: juniper_official/Security-Compliance/check-interface-to-disable.rule [hunk 17]
```diff
@@ -0,0 +1,101 @@
+/*
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-to-disable.rule:1]: Made the changes for all rules.

File: juniper_official/Security-Compliance/check-interface-to-disable.rule [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+            /*
+             * Anomaly detection logic.
+             */
+            trigger interfaces-to-disable {
+                frequency 2offset;
+                term disable-interfaces {
+                    then {
+                        status {
+                            color red;
+                            message "The list of interfaces that do not have configuration and can be diabled.$list-of-interfaces-make-down";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-to-disable.rule:29]: Updated to match only ethernet interfaces.

File: juniper_official/Security-Compliance/check-interface-protocols.rule [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+            trigger ospf_interface_all {
+                frequency 2offset;
+                term ospf-interface-all-present {
+                    when {
+                        matches-with "$ospfInterfaceName" all;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "ospf interface all should not be set.";
```
[the remark this answers is not in the document]
Comment (vvikramb) [reply juniper_official/Security-Compliance/check-interface-protocols.rule]: Checking if "interface all" is enabled for ospf protocols and raising alarm if it is set.

**Review by gvarao — APPROVED** [review]

**Pull request merged**